src/chunk_constraint.c

```diff
@@ -899,3 +907,38 @@ chunk_constraint_rename_hypertable_constraint(int32 chunk_id, const char *oldnam
 													  RowExclusiveLock,
 													  CurrentMemoryContext);
 }
+
+static bool
+chunk_constraint_get_name_from_hypertable_tuple(TupleInfo *ti, void *data)
+{
+	GetNameFromHypertableConstraintInfo *info = data;
+
+	bool		nulls[Natts_chunk_constraint];
+	Datum		values[Natts_chunk_constraint];
+
+
+	heap_deform_tuple(ti->tuple, ti->desc, values, nulls);
+	info->chunk_constraint_name = DatumGetName(values[Anum_chunk_constraint_constraint_name - 1]);
```

Comment:
Isn't this the place to `pstrdup()` rather than the calling function since `values` is on the stack (although I guess the `Name` itself might be on some other memory context that the tuple is on)? Otherwise I don't see the need for the `pstrdup`.